lib/recurly/client.rb

```diff
@@ -118,18 +119,59 @@ def delete(path, **options)
     # @return [Logger]
     attr_reader :logger
 
-    def run_request(request, headers)
-      read_headers @conn.run_request(request.method, request.path, request.body, headers)
+    def connection_pool
+      @@connection_pool ||= Recurly::ConnectionPool.new
     end
 
-    def handle_response!(request, faraday_resp)
-      response = HTTP::Response.new(faraday_resp, request)
-      raise_api_error!(response) unless (200...300).include?(response.status)
+    def run_request(request, options = {})
+      request["Accept"] = "application/vnd.recurly.#{api_version}".chomp # got this method from operations.rb
+      request["Authorization"] = "Basic #{Base64.encode64(@api_key)}".chomp
+      request["User-Agent"] = "Recurly/#{VERSION}; #{RUBY_DESCRIPTION}"
+
+      # TODO this is undocumented until we finalize it
+      options[:headers].each { |header, v| request[header] = v } if options[:headers]
+
+      connection_pool.with_connection do |http|
+        http.open_timeout = options[:open_timeout] || 20
+        http.read_timeout = options[:read_timeout] || 60
+
```

Comment:
We need to plug in the logger somewhere for debugging purposes. If we drop it here, that will work. I used this code:

```
http.set_debug_output(logger)
```

Reply:
```
/app/clients/ruby/lib/recurly/client.rb:137: warning: Net::HTTP#set_debug_output called after HTTP started
```

Perhaps it can only be set only once per started instance.

Reply:
Also, we need to gate this only for certain logger levels.

Reply:
I added back the logger and added specs to make sure it gets called with the correct levels.